onnx/version_converter/convert.h

```diff
@@ -740,6 +740,10 @@ class DefaultVersionConverter : public BaseVersionConverter {
         OpSetID(15), OpSetID(16)));
       registerAdapter(make_unique<CompatibleAdapter>("PRelu",
         OpSetID(15), OpSetID(16)));
+    
+      /******** 16 -> 17 ********/
+      registerAdapter(make_unique<CompatibleAdapter>("NonZero", OpSetID(16), OpSetID(17)));
```

Comment:
This looks like a tricky edge case, since the behavior is not really preserved for the scalar case. If the input is known to be a scalar, we could fail or produce a warning, I suppose. But if the input rank is not known, unclear what we can do.